src/table/components/virtualized-table/TableContainer.tsx

```diff
@@ -21,26 +22,7 @@ const TableContainer = (props: TableContainerProps) => {
   const { width } = useColumnSize(rect, columns, headerStyle, bodyStyle);
   const totalWidth = columns.reduce((prev, curr, index) => prev + width[index], 0);
   const [totalHeight, setTotalHeight] = useState(pageInfo.rowsPerPage * DEFAULT_ROW_HEIGHT);
-
-  const onScrollHandler = (event: React.SyntheticEvent) => {
-    if (headerRef.current) {
-      headerRef.current.scrollTo(event.currentTarget.scrollLeft);
-    }
-
-    if (bodyRef.current) {
-      bodyRef.current.scrollTo({
-        scrollLeft: event.currentTarget.scrollLeft,
-        scrollTop: event.currentTarget.scrollTop,
-      });
-    }
-
-    if (innerForwardRef.current) {
-      // Keep full size container in sync with the height calculation in react-window is doing
-      if (totalHeight !== innerForwardRef.current.clientHeight) {
-        setTotalHeight(innerForwardRef.current.clientHeight);
-      }
-    }
-  };
+  const scrollHandler = useScrollHandler(headerRef, bodyRef, innerForwardRef, totalHeight, setTotalHeight);
```

Comment:
Sry about this one, does not really have anything to do with constraints. I ended up doing a change to it as I tought it would be needed, but then I didn't need it. I still want to keep the refactoring part of my change.

There should be no change other than refactoring and added tests for the scroll handler.

Reply:
np, thanks for explaining